st2actions/st2actions/runners/ssh/paramiko_ssh.py

```diff
@@ -579,13 +579,39 @@ def _connect(self, host, socket=None):
         if socket:
             conninfo['sock'] = socket
 
+        conninfo = self._get_conninfo_from_ssh_config(conninfo)
         client = paramiko.SSHClient()
         client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
         client.connect(**conninfo)
 
         return client
 
     @staticmethod
+    def _get_conninfo_from_ssh_config(conninfo):
```

Comment:
A minor thing that's concerning is that you are passing an object and mutating it inside a static method. Plus the method is called _get_conninfo... 

Should we refactor this to something like:

```
if cfg.CONF.ssh_runner.use_ssh_config:
  conninfo_ssh_config = self._get_conninfo_from_ssh_config_for_host(host)
  conninfo.update(conninfo_ssh_config)
```